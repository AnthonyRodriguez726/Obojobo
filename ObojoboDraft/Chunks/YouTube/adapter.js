let Adapter = {
	construct(model, attrs) {
		if (__guard__(attrs != null ? attrs.content : undefined, x => x.videoId) != null) {
			return (model.modelState.videoId = attrs.content.videoId)
		} else {
			return (model.modelState.videoId = null)
		}
	},

	clone(model, clone) {
		return (clone.modelState.videoId = model.modelState.videoId)
	},

	toJSON(model, json) {
		return (json.content.videoId = model.modelState.videoId)
	},

	toText(model) {
		return `https://www.youtube.com/watch?v=${model.modelState.videoId}`
	}
}

export default Adapter
function __guard__(value, transform) {
	return typeof value !== 'undefined' && value !== null ? transform(value) : undefined
}
