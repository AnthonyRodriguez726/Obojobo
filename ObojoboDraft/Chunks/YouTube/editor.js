import React from 'react'

const YOUTUBE_NODE = 'ObojoboDraft.Chunks.YouTube'

class Node extends React.Component {
	constructor(props) {
		super(props);
		this.state = this.props.node.data.get('content')
	}

	handleTypeChange(event){
		const editor = this.props.editor
		const change = editor.value.change()

		this.setState({ videoId: event.target.value })

		change.setNodeByKey(this.props.node.key, { data: { content: {
			videoId: event.target.value
		}}})
		editor.onChange(change)
	}

	renderInput() {
		return (
			<input
				value={this.state.videoId}
				onChange={event => this.handleSourceChange(event)}
				onClick={event => event.stopPropagation()} />
		)
	}

	renderFrame() {
		const { isFocused } = this.props

		const wrapperStyle = {
			position: 'relative',
			outline: isFocused ? '2px solid blue' : 'none',
		}

		const maskStyle = {
			display: isFocused ? 'none' : 'block',
			position: 'absolute',
			top: '0',
			left: '0',
			height: '100%',
			width: '100%',
			cursor: 'cell',
			zIndex: 1,
		}

		const iframeStyle = {
			display: 'block',
		}

		return (
			<div style={wrapperStyle}>
				<div style={maskStyle} />
				<iframe
					id="ytplayer"
					type="text/html"
					width="640"
					height="476"
					src={'https://www.youtube.com/embed/' + this.state.videoId}
					frameBorder="0"
					style={iframeStyle}
				/>
			</div>
		)
	}

	render() {
		const { isSelected } = this.props

		return (
			<div {...this.props.attributes}>
				{this.renderFrame()}
				{isSelected ? this.renderInput() : null}
			</div>
		)
	}
}

const insertNode = change => {
	console.log('inserting')
	change
		.insertBlock({
			type: YOUTUBE_NODE,
			data: { content: { videoId: 'FaHEusBG20c' } },
			isVoid: true
		})
		.collapseToStartOfNextText()
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content') || {}

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case YOUTUBE_NODE:
				return <Node {...props} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.YouTube': {
				isVoid: true,
			}
		}
	}
}

const YouTube = {
	components: {
		Node,
	},
	helpers: {
		insertNode,
		slateToObo
	},
	plugins
}

export default YouTube
