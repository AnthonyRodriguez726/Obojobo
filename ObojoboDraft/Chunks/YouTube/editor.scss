/* stylelint-disable */
@import '../../../../../../../scss/includes';

.component[data-component-type='ObojoboDraft.Chunks.YouTube'] {
	&.selection-contains,
	&.selection-inside,
	&.selection-start,
	&.selection-end {
		.obojobo-draft--components--delete-button {
			display: block;
		}
	}
}

.obojobo-draft--chunks--you-tube {
	$color-blank: #000000;

	.anchor {
		z-index: 2;
	}

	> .container {
		height: 25em;
		margin: 0 auto;
		position: relative;
		width: 32em;
	}

	.placeholder,
	.edit,
	img,
	iframe {
		height: 100%;
		margin: 0 auto;
		width: 100%;
	}

	.placeholder {
		@include placeholder(25rem);
	}

	.edit {
		background: $color-blank;
		position: relative;
		z-index: 3;
	}

	.obojobo-draft--components--delete-button {
		display: none;
		position: absolute;
		right: 0;
		top: 0;
		transform: translate(50%, -50%);
		z-index: 3;
	}

	.obojobo-draft--components--edit-button {
		position: absolute;
		right: 1em;
		top: 1em;
		z-index: 3;
	}
}
/* stylelint-enable */
