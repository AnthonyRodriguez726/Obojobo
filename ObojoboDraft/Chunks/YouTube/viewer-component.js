import './viewer-component.scss';

import Common from 'Common'
let { OboComponent } = Common.components;

export default class YouTube extends React.Component {
	render() {
		let data = this.props.model.modelState;

		return <OboComponent model={this.props.model} moduleData={this.props.moduleData}>
			<div className='obojobo-draft--chunks--you-tube viewer'>
				<iframe src={`https://www.youtube.com/embed/${data.videoId}`} frameBorder="0" allowFullScreen="true" />
			</div>
		</OboComponent>;
	}
}