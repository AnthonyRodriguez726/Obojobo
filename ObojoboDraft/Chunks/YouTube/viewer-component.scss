@import '~styles/includes';

.obojobo-draft--chunks--you-tube {
	display: block;
	margin: 1em auto;
	text-align: center;
	position: relative;

	iframe {
		height: 25em;
		margin: 0 auto;
		width: 32em;
	}

	input {
		width: 100%;
	}
}
