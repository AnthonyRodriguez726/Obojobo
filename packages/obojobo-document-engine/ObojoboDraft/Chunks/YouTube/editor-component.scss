@import '~styles/includes';

.obojobo-draft--chunks--you-tube {
	.obojobo-draft--components--button {
		position: absolute;
		top: 1em;
		right: 7em;

		button {
			border: 1px solid $color-bg;
		}
	}

	.empty-frame {
		height: 25em;
		margin: auto;
		width: 32em;
		border: 1px solid lighten($color-action, 9%);
		border-radius: $dimension-rounded-radius;

		div {
			line-height: 25em;
		}
	}
}
