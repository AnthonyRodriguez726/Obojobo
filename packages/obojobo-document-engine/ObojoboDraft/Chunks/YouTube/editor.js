import React from 'react'
import { Block } from 'slate'

import emptyNode from './empty-node.json'
import Icon from './icon'

const YOUTUBE_NODE = 'ObojoboDraft.Chunks.YouTube'

class Node extends React.Component {
	constructor(props) {
		super(props)
	}

	handleSourceChange(event) {
		const editor = this.props.editor
		const change = editor.value.change()

		change.setNodeByKey(this.props.node.key, {
			data: {
				content: {
					videoId: event.target.value
				}
			}
		})
		editor.onChange(change)
	}

	render() {
		const { isFocused } = this.props
		const content = this.props.node.data.get('content')

		const maskStyle = {
			display: isFocused ? 'none' : 'block',
			position: 'absolute',
			top: '0',
			left: '0',
			height: '100%',
			width: '100%',
			cursor: 'cell',
			zIndex: 1
		}

		return (
			<div className={'obojobo-draft--chunks--you-tube viewer pad'}>
				<div style={maskStyle} />
				<iframe
					src={'https://www.youtube.com/embed/' + content.videoId}
					frameBorder="0"
					allowFullScreen="true"
				/>
				<input
					value={content.videoId}
					aria-label={'Youtube Video Id'}
					onChange={event => this.handleSourceChange(event)}
					onClick={event => event.stopPropagation()}
				/>
			</div>
		)
	}
}

const insertNode = change => {
	change
		.insertBlock(Block.fromJSON(emptyNode))
		.focus()
}

const slateToObo = node => {
	const json = {}
	json.id = node.key
	json.type = node.type
	json.content = node.data.get('content') || {}
	json.children = []

	return json
}

const oboToSlate = node => {
	const json = {}
	json.object = 'block'
	json.key = node.id
	json.type = node.type
	json.data = { content: node.content }

	return json
}

const plugins = {
	renderNode(props) {
		switch (props.node.type) {
			case YOUTUBE_NODE:
				return <Node {...props} {...props.attributes} />
		}
	},
	schema: {
		blocks: {
			'ObojoboDraft.Chunks.YouTube': {
				isVoid: true
			}
		}
	}
}

const YouTube = {
	name: YOUTUBE_NODE,
	components: {
		Node,
		Icon
	},
	helpers: {
		insertNode,
		slateToObo,
		oboToSlate
	},
	json: {
		emptyNode
	},
	plugins
}

export default YouTube
