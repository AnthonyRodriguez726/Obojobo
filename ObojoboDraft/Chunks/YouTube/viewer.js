import Common from 'Common'
import adapter from './adapter'
import ViewerComponent from './viewer-component'

let SelectionHandler = Common.chunk.focusableChunk.FocusableSelectionHandler

Common.Store.registerModel('ObojoboDraft.Chunks.YouTube', {
	type: 'chunk',
	adapter: adapter,
	componentClass: ViewerComponent,
	selectionHandler: new SelectionHandler()
})
