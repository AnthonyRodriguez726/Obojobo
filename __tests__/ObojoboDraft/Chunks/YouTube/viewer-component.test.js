import React from 'react'
import renderer from 'react-test-renderer'

import YouTube from '../../../../ObojoboDraft/Chunks/YouTube/viewer-component'
import OboModel from '../../../../__mocks__/_obo-model-with-chunks'

describe('YouTube', () => {
	let model = OboModel.create({
		id: 'id',
		type: 'ObojoboDraft.Chunks.YouTube',
		content: {
			videoId: 'dQw4w9WgXcQ'
		}
	})

	let moduleData = {
		focusState: {}
	}

	test('YouTube component', () => {
		const component = renderer.create(
			<YouTube model={model} moduleData={moduleData} />
		)
		let tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})