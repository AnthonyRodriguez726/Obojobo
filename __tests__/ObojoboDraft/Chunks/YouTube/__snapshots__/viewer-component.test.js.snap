// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube YouTube component 1`] = `
<div
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.YouTube"
  id="obo-id"
  tabIndex={-1}
>
  <div
    className="obojobo-draft--chunks--you-tube viewer"
  >
    <iframe
      allowFullScreen="true"
      frameBorder="0"
      src="https://www.youtube.com/embed/dQw4w9WgXcQ"
    />
  </div>
</div>
`;
